dev/integration_tests/android_splash_screens/splash_screen_kitchen_sink/android/app/src/main/java/io/flutter/splash_screen_kitchen_sink/MainActivity.java

```diff
@@ -0,0 +1,190 @@
+package io.flutter.splash_screen_kitchen_sink;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewTreeObserver;
+
+import androidx.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.android.FlutterView;
+import io.flutter.embedding.android.SplashScreen;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.StringCodec;
+import io.flutter.view.FlutterMain;
+
+public class MainActivity extends FlutterActivity {
+  private static final String TAG = "MainActivity";
+
+  private static FlutterEngine flutterEngine;
+
+  static {
+    // Explicitly activates Debug logging for the Flutter Android embedding.
+    io.flutter.Log.setLogLevel(Log.DEBUG);
+  }
+
+  // Sends a JSON-serialized log of test events from Android to Flutter.
+  private BasicMessageChannel<String> testChannel;
+  // Log of splash events that is updated, serialized, and sent to Flutter.
+  private SplashTestLog splashTestLog;
+
+  /**
+   * We explicitly provide a {@code FlutterEngine} so that every rotation does not create a
+   * new FlutterEngine. Creating a new FlutterEngine on every orientation would cause the
+   * splash experience to restart upon every orientation change, which is not what we're
+   * interested in verifying in this example app.
+   */
+  @Override
+  public FlutterEngine provideFlutterEngine(Context context) {
+    if (flutterEngine == null) {
+      flutterEngine = new FlutterEngine(context);
+
+      flutterEngine.getDartExecutor().executeDartEntrypoint(new DartExecutor.DartEntrypoint(
+          getAssets(),
+          FlutterMain.findAppBundlePath(context),
+          "main"
+      ));
+
+      // Setup the channel that sends splash test log updates from Android to Flutter.
```

Comment:
Since this is the first place where this concept appears, describe what this is for (or put it in the readme)

Reply:
I added to the README with details about it.